src/mediastreambroker/DeviceControllerBasedMediaStreamBroker.ts

```diff
@@ -2,10 +2,21 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import DeviceController from '../devicecontroller/DeviceController';
+import EventController from '../eventcontroller/EventController';
 import MediaStreamBroker from '../mediastreambroker/MediaStreamBroker';
 
 export default interface DeviceControllerBasedMediaStreamBroker
   extends DeviceController,
     MediaStreamBroker {
   // This interface combines a device controller with a media stream broker
+
+  /**
+   * Set the current event controller for publishing events
+   */
+  setEventController?(eventController: EventController): void;
```

Comment:
Why are these get set an optional APIs?

Reply:
We could make it required. I was thinking if a builder has their own DeviceController that they don't want to use the EventController for they could just not implement these.